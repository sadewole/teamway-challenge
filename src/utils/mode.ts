/**
 * @param ['introvert', 'introvert', 'extrovert']
 * @returns 'introvert'
 */
export const mode = (arr: string[]) => {
  return arr
    .sort(
      (a, b) =>
        arr.filter((v) => v === a).length - arr.filter((v) => v === b).length,
    )
    .pop()
}
