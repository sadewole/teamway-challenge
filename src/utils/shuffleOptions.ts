import { QuestionOptionT } from '../context/types'

export const shuffleOptions = (arr: QuestionOptionT[]) =>
  arr
    .map((value) => ({ value, sort: Math.random() }))
    .sort((a, b) => a.sort - b.sort)
    .map(({ value }) => value)
