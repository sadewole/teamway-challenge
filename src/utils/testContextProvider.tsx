import { PropsWithChildren, ReactElement } from 'react'
import { BrowserRouter } from 'react-router-dom'
import { render, RenderOptions } from '@testing-library/react'

import {
  QuestionContext,
  State as QuestionState,
} from '../context/QuestionProvider'
import { AccountContext } from '../context/AccountProvider'

import { UsernameT } from '../context/types'

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  accountProviderValue?: UsernameT | null
  questionProviderValue?: QuestionState
}

const customRender = (
  ui: ReactElement,
  {
    accountProviderValue = null,
    questionProviderValue = {
      questions: [],
      currentQuestionIndex: 0,
      currentQuestion: null,
      selectedAnswers: {},
    },
    ...renderOptions
  }: ExtendedRenderOptions = {},
) => {
  const Wrapper = ({ children }: PropsWithChildren<{}>) => {
    return (
      <AccountContext.Provider value={accountProviderValue}>
        <QuestionContext.Provider value={questionProviderValue}>
          <BrowserRouter>{children}</BrowserRouter>
        </QuestionContext.Provider>
      </AccountContext.Provider>
    )
  }

  return render(ui, { wrapper: Wrapper, ...renderOptions })
}

export { customRender }
