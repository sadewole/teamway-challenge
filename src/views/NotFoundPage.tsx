import { NavLink } from 'react-router-dom'

const NotFoundPage = () => {
  return (
    <div className="flex flex-col items-center justify-center -mt-24 h-screen">
      <h1 className="text-3xl font-semibold mb-2">Page not found</h1>
      <p>Sorry, the URL you have used may be incorrect.</p>
      <p>
        Check the spelling or{' '}
        <NavLink to="/" className="font-semibold text-blue-500 underline">
          click here
        </NavLink>{' '}
        to navigate to the homepage.
      </p>
    </div>
  )
}

export default NotFoundPage
