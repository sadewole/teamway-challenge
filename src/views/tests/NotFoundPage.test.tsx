import { customRender } from '../../utils/testContextProvider'
import NotFoundPage from '../NotFoundPage'

it('Page should match snapshot', () => {
  const { container } = customRender(<NotFoundPage />)
  expect(container).toMatchSnapshot()
})
