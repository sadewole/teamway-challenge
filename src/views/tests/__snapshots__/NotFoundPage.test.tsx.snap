// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should match snapshot 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center -mt-24 h-screen"
  >
    <h1
      class="text-3xl font-semibold mb-2"
    >
      Page not found
    </h1>
    <p>
      Sorry, the URL you have used may be incorrect.
    </p>
    <p>
      Check the spelling or
       
      <a
        aria-current="page"
        class="font-semibold text-blue-500 underline active"
        href="/"
      >
        click here
      </a>
       
      to navigate to the homepage.
    </p>
  </div>
</div>
`;
